components/bl-leafletMap-component/src/fullscreen-button.js

```diff
@@ -0,0 +1,30 @@
+import { useCallback, useState, useEffect } from 'react';
+
+import { TurnOnFullscreenIcon } from './icons';
+import { TurnOffFullscreenIcon } from './icons';
+
+export function FullscreenButton({ fullscreen, eventHandlers }) {
+  const { onFullscreenClick } = eventHandlers;
+
+  const [currentFullscreen, setCurrentFullscreen] = useState(null);
+
+  useEffect(() => {
+    if (fullscreen !== undefined) {
+      setCurrentFullscreen(fullscreen);
+    }
+  }, [fullscreen]);
+
+  const handleClick = useCallback(() => {
+    onFullscreenClick({ fullscreen: !currentFullscreen });
```

Comment:
does this mean in order to toggle fullscreen the developer has to change the `component.fullscreen` property manually in the `onFullscreenClick ` handler ?

Reply:
In order for us to be able to control the fullscreen property in logic, we pass the current fullscreen value to the event and then replace the fullscreen property with this value

Reply:
who will replace the property? the customer or does it happen automatically in the code?